client/packages/host/src/api/api.ts

```diff
@@ -22,8 +29,57 @@ export const getHostQueries = (sdk: Sdk) => ({
   },
   initialise: async (settings: SyncSettingsInput) => {
     const result = await sdk.initialiseSite({
-      syncSettings: settings,
+      syncSettings: cleanSyncSettings(settings),
+    });
+    return result.initialiseSite;
+  },
+  update: async (settings: SyncSettingsInput) => {
+    const result = await sdk.updateSyncSettings({
+      syncSettings: cleanSyncSettings(settings),
     });
-    return result?.initialiseSite;
+    return result?.updateSyncSettings;
   },
 });
+
+// In typescript it's allowed to have excess properties on object
+// to avoid errors thrown in mutation, we should remove any excess properties for input
+function cleanSyncSettings({
+  username,
+  password,
+  url,
+  intervalSeconds,
+}: SyncSettingsInput): SyncSettingsInput {
+  return { username, password, url, intervalSeconds };
+}
+
+export function mapSyncError(
+  t: TypedTFunction<LocaleKey>,
+  error: SyncErrorsFragment,
+  defaultKey?: LocaleKey
+): ErrorWithDetailsProps {
+  if (error.__typename === 'UnknownSyncError') {
+    return {
+      error: defaultKey ? t(defaultKey) : t('error.unknown-sync-error'),
+      details: error.fullError,
+    };
+  }
+
+  const errorMapping: { [key in SyncErrorVariant]: LocaleKey } = {
```

Comment:
Hard coding the mappings like this introduces a potential failure - when the server enum is extended and this mapping isn't. I don't see what benefit it gives over the previous approach which is that the server supplies a code and that code is used as the translation string lookup. If the lookup fails, a generic error is shown the user.

what happens in this implementation when `SyncErrorVariant.NewEnumValue` is returned by the server?

A suggestion is to have a default, if the error mapping fails to match a value:

```
  const errorKey = errorMapping[error.errorVariant] ?? 'error.something-wrong';
  return { error: t(errorKey), details: error.fullError };
```

you can also provide a default when calling the typedTFunction : `t('some-key', 'default value')`

Reply:
With graphql types like this, when api changes front end fails to compile (`generate` must of-course run)

Reply:
yeah, that being a manual process it can't be always relied upon - though it will show a problem next time someone generates and tries to compile. I still like the loose linking, lack of code duplication and fallback 🤷

Reply:
> I still like the loose linking, lack of code duplication and fallback

Personally I am very strong on strong typing (and from memory Josh was too, [he is the one suggested union error return in the first place](https://github.com/openmsupply/remote-server/pull/359#issue-1004161704)), if we don't strong type then we would possible get visible and invisible runtime errors.

> that being a manual process it can't be always relied upon

We can make it safe with action test. And also i think manual process here can be automated and even if it wasn't automated I think it's much better then manual process of testing all combinations of errors and their expected output.

>  fallback

Added fallback 7f9a94007785805f6631a89bdf766b02b24c2976, but I think it would introduce invisible error (without fallback we would get `undefined` with it will be default)

> lack of code duplication 

You mean the duplication here with Variants being mapped to LocaleKeys ? Sure but isn't it safer (consider someone saw that `_` was previously being used in errors, and decided to change it back to `-`, wouldn't it be great if compiler will test this rather then us testing it manually ? 

I'll try to add a discussion, to see our stance on strong typing, maybe we can vote/discuss to see when we can be loose with types.